web_api/core/test_views.py

```diff
@@ -333,6 +700,651 @@ def test_accounts(authed_client: Client, user: User) -> None:
     )
 
 
+@pytest.fixture
+def patch_start_trial(mocker: Any) -> None:
+    fake_customer = stripe.Customer.construct_from(
+        dict(object="customer", id="cust_Gx2a3gd5x6",), "fake-key",
+    )
+    mocker.patch("core.models.stripe.Customer.create", return_value=fake_customer)
+    mocker.patch("core.models.stripe.Customer.modify", return_value=fake_customer)
+
+
+@pytest.mark.django_db
+def test_start_trial(
+    authed_client: Client, user: User, patch_start_trial: object
+) -> None:
+    """
+    When a user starts a trial we should update their account.
+    """
+    account = Account.objects.create(
+        github_installation_id=377930,
+        github_account_id=900966,
+        github_account_login=user.github_login,
+        github_account_type="User",
+    )
+    AccountMembership.objects.create(account=account, user=user, role="member")
+    assert account.trial_start is None
+    assert account.trial_expiration is None
+    assert account.trial_started_by is None
+    assert (
+        account.trial_expired() is False
+    ), "when a trial is inactive, it shouldn't have expired."
+    res = authed_client.post(
+        f"/v1/t/{account.id}/start_trial", dict(billingEmail="b.lowe@example.com")
+    )
+    assert res.status_code == 204
+    account.refresh_from_db()
+    assert account.trial_start is not None
+    assert (
+        (account.trial_start - account.trial_expiration).total_seconds()
+        - datetime.timedelta(days=14).total_seconds()
+        < 60 * 60
+    ), "times should be within an hour of each other. This should hopefully avoid flakiness around dates."
+    assert account.trial_started_by == user
+    assert account.trial_expired() is False
+    assert account.trial_email == "b.lowe@example.com"
+
+
+def equal_dates(a: datetime.datetime, b: datetime.datetime) -> bool:
```

Comment:
```suggestion
def similar_dates(a: datetime.datetime, b: datetime.datetime) -> bool:
```

or maybe `close_dates` since they aren't actually equal